docs/en/stack/ml/anomaly-detection/ml-limitations.asciidoc

```diff
@@ -70,6 +70,24 @@ You cannot use the following field names in the `by_field_name` or
 `over_field_name` properties in a job: `by`; `count`; `over`. This limitation
 also applies to those properties when you create advanced jobs in {kib}.
 
+
+[discrete]
+[[ml-arrays-limitations]]
+=== Arrays in analyzed fields are turned into comma-separated strings
+
+If an {anomaly-job} is configured to analyze a field that contains an array, 
+then the array is turned into a comma-separated concatenated string. The items 
+in the array are sorted alphabetically and the duplicated items are removed. For 
+example, the array `["zebra", "dog", "cat", "alligator", "cat"]` becomes 
+`alligator,cat,dog,zebra`. The Anomaly Explorer charts does not display results 
```

Comment:
This is actually a bit of a nightmare.

From my reading of the code in `SourceField` and `DocValueField`, my understanding is that _if_ an array field is aggregatable, i.e. has mappings that imply doc values will be stored, then what is described will happen: the distinct values will be put into the comma separated list in alphabetical order.  However, if an array field is not aggregatable and is retrieved from `_source` then they won't.  So in the example given the comma separated string would be `zebra,dog,cat,alligator,cat`.

Do you agree @dimitris-athanasiou?

Reply:
I have just tested this and it is exactly as you describe it.

Reply:
😬 

I guess we have to document what happens today in the short term.  But maybe in the longer term we should make both cases work the same.

Reply:
Thanks for the clarification and the test! @droberts195 I modified the documentation accordingly via 
c354f86.